### PR TITLE
dot, dot/state, lib/genesis: migrate genesis methods

### LINKED ISSUE #720: migrate dot genesis methods to genesis
## Expected Behavior

The dot package should not include genesis methods.

## Current Behavior

The genesis package should include all genesis methods currently in the dot package.

### PR DESCRIPTION
**Pull request by ryanchristo**
## Changes
- migrate dot genesis methods to genesis package and clean up node and state initialization

## Tests:
```
go test ./... -short
```

### Issues:
closes #720

### CODE REVIEW

File: dot/state/service.go [hunk 1]
```diff
@@ -66,70 +67,99 @@ func (s *Service) DB() database.Database {
 // Initialize initializes the genesis state of the DB using the given storage trie. The trie should be loaded with the genesis storage state.
 // The trie does not need a backing DB, since the DB will be created during Service.Start().
```
Comment (noot) [inline dot/state/service.go:68]: remove this line

Comment (ryanchristo) [reply dot/state/service.go]: done :+1:

**Comment by ryanchristo** [comment]
cc @noot can I get a second review? cc @thomasmodeneis and/or @edwardmack this pull request is ready for review when you have a moment. :pray:

**Comment by noot** [comment]
I updated the genesis file to set `name` as `noot` then ran:

```
elizabeth@elizabeth:~/go/src/github.com/ChainSafe/gossamer$ ./bin/gossamer --key alice init
INFO[04-08|11:51:45] [cmd] Loading node implementation...     name=gssmr
WARN[04-08|11:51:45] [cmd] genesis mismatch, overwriting name name=noot
```

but when I run the node I get:
```
elizabeth@elizabeth:~/go/src/github.com/ChainSafe/gossamer$ ./bin/gossamer --key alice 
INFO[04-08|11:52:21] [cmd] Loading node implementation...     name=gssmr
INFO[04-08|11:52:21] [dot] Creating node services...          datadir=/home/elizabeth/.gossamer/gssmr
INFO[04-08|11:52:21] [dot] Creating state service... 
INFO[04-08|11:52:21] [dot] Creating core service...           authority=true
INFO[04-08|11:52:22] [babe] config                            SlotDuration (ms)=1000 EpochLength (slots)=6
INFO[04-08|11:52:22] [dot] Creating network service...        roles=4 port=7001 bootnodes=[] protocol=/gossamer/gssmr/0 nobootstrap=false nomdns=false
WARN[04-08|11:52:22] [network] Bootstrap is enabled but no bootstrap nodes are defined 
INFO[04-08|11:52:22] [cmd] Starting node...                   name=gssmr
INFO[04-08|11:52:22] [dot] Starting node services... 
```

doesn't look like the name is getting update properly?

**Comment by ryanchristo** [comment]
I believe this was not an issue with the `setDotConfigFromGenesis` method included in the main action command that was remover per request but I will take another look.

File: dot/state/service.go [hunk 2]
```diff
@@ -160,15 +192,15 @@ func (s *Service) Start() error {
...
 	if err != nil {
-		return fmt.Errorf("cannot get latest hash: %s", err)
+		return fmt.Errorf("failed to get best block hash: %s", err)
 	}
 
 	log.Trace("[state] start", "best block hash", fmt.Sprintf("0x%x", bestHash))
 
 	// create storage state
-	s.Storage, err = NewStorageState(db, trie.NewEmptyTrie())
+	s.Storage, err = NewStorageStateFromTrie(db, trie.NewEmptyTrie())
```
Comment (noot) [inline dot/state/service.go:201]: why is this renamed? can you change it back to `NewStorageState`

Comment (ryanchristo) [reply dot/state/service.go:201]: It matches the naming convention of the other helper methods and it is important information to know that the storage state is being create from the genesis trie, no?

I can change it back but I think this is an improvement.

Comment (noot) [reply dot/state/service.go:201]: well, since there is not `NewStorageState` function, I think it's a bit confusing now. `FromTrie` implies there is a `NewStorageState` without trie

Comment (ryanchristo) [reply dot/state/service.go:201]: Ok, no problem I will change it back if this is more confusing to you.
